src/fidesops/util/encryption/secrets_util.py

```diff
@@ -0,0 +1,72 @@
+import logging
+import secrets
+from typing import TypeVar, Optional, Set, List, Union
+
+from fidesops.schemas.masking.masking_secrets import (
+    MaskingSecretMeta,
+    SecretDataType,
+    MaskingSecretCache,
+)
+from fidesops.util.cache import get_masking_secret_cache_key, get_cache
+
+T = TypeVar("T")
+logger = logging.getLogger(__name__)
+
+
+class SecretsUtil:
+    @staticmethod
+    def get_or_generate_secret(
+        privacy_request_id: Optional[str], masking_secret_meta: MaskingSecretMeta[T]
+    ) -> T:
+        if privacy_request_id is not None:
+            secret: T = SecretsUtil._get_secret_from_cache(
+                privacy_request_id, masking_secret_meta
+            )
+            if not secret:
+                logger.warning(
+                    f"Secret type {masking_secret_meta.secret_type} expected from cache but was not present for masking strategy {masking_secret_meta.masking_strategy}"
+                )
+                return masking_secret_meta.generate_secret()
+                # todo- should we attempt to re-cache in this case?
```

Comment:
If for some reason secret is not in cache, we can generate it on the fly. Should we also attempt to re-cache?